apps/authentication/api/mfa.py

```diff
@@ -74,42 +78,41 @@ def _update_context_with_error(self, context, error_message):
 
     def _update_cache(self, context):
         cache_key = self.get_face_cache_key(context['token'])
-        cache.set(cache_key, context, MFA_FACE_CONTEXT_CACHE_TTL)
+        cache.set(cache_key, context, FACE_CONTEXT_CACHE_TTL)
 
     def _handle_success(self, context, face_code):
         context.update({
             'is_finished': True,
             'success': True,
             'face_code': face_code
         })
+        action = context.get('action', None)
+        if action == 'login_asset':
+            with tmp_to_root_org():
+                connection_token_id = context.get('connection_token_id')
+                token = ConnectionToken.objects.filter(id=connection_token_id).first()
+                token.is_active = True
+                token.save()
         self._update_cache(context)
 
 
 class MFAFaceContextApi(AuthMixin, RetrieveAPIView, CreateAPIView):
     permission_classes = (AllowAny,)
-    face_token_session_key = MFA_FACE_SESSION_KEY
+    face_token_session_key = FACE_SESSION_KEY
 
     @staticmethod
     def get_face_cache_key(token):
-        return f"{MFA_FACE_CONTEXT_CACHE_KEY_PREFIX}_{token}"
+        return f"{FACE_CONTEXT_CACHE_KEY_PREFIX}_{token}"
 
     def new_face_context(self):
-        token = uuid.uuid4().hex
-        cache_key = self.get_face_cache_key(token)
-        face_context = {
-            "token": token,
-            "is_finished": False
-        }
-        cache.set(cache_key, face_context, MFA_FACE_CONTEXT_CACHE_TTL)
-        self.request.session[self.face_token_session_key] = token
-        return token
+        return self.create_face_verify_context()
 
     def post(self, request, *args, **kwargs):
         token = self.new_face_context()
         return Response({'token': token})
 
     def get(self, request, *args, **kwargs):
-        token = self.request.session.get('mfa_face_token')
+        token = self.request.session.get(self.face_token_session_key)
 
         cache_key = self.get_face_cache_key(token)
         context = cache.get(cache_key)
```

Comment:
The main difference between `perform_create` method in two versions is the usage of `tmp_to_root_org()` inside `_handle_success`. Also, there seems to be an error message "missing field 'face_code'" that needs to be handled differently.

In version one, this part:

```
context.update({
                    'is_finished': True,
                    'success': True,
                    'error': str(error),
                 })
```

Should read: 

```python
context.update({  
           'is_finished': True,
           'success': True,
           'error': '',
       })
```

This would correct for what appears to be a missing string literal value being passed into `_handle_success`, which results in the exception handling logic failing when attempting to set or update keys within the dictionary.

On another note, it may add utility to wrap the verification process with proper logging and response status codes. This could help catch unexpected errors before they reach the frontend user interface.

Additionally, make sure all calls to methods like _update_cache() have a clear intention on setting state back to expected behavior. It might also enhance readability by having consistent capitalization across variables names in different environments (`__class_name_upper_case__, etc...`). The implementation itself looks good though!